book/doctrine.rst

```diff
@@ -793,6 +793,16 @@ For more information on Doctrine's Query Builder, consult Doctrine's
 
 .. _book-doctrine-custom-repository-classes:
 
+Query Builder vs DQL
+~~~~~~~~~~~~~~~~~~~~
+
+The ``QueryBuilder`` rather to use instead of ``DQL`` when query logic based
+on some result of custom conditions. It allows you to avoid a query string
+concatenating manually. The ``QueryBuilder`` do this concatenation by itself
+while you calling suitable methods. Also, with ``QueryBuilder`` you could easy
+take away some parts of query to separate methods and reuse it in other queries.
+In other cases you probably should use a ``DQL`` syntax to build your queries.
```

Comment:
I suggest to reword the paragraph to something like this:

> When the actual query depends on dynamic conditions, your code soon becomes hard to read as you start to concatenate strings. In these cases, it's better to use the Doctrine QueryBuilder which provides a fluent interface to dynamically add every part of the query.

What do you think?